src/CodeGeneration.Roslyn/DocumentTransform.cs

```diff
@@ -240,5 +206,75 @@ private static string GetFullTypeName(INamedTypeSymbol symbol)
 
             return nameBuilder.ToString();
         }
+
+        private class EnrichingCodeGeneratorProxy : IRichCodeGenerator
+        {
+            public EnrichingCodeGeneratorProxy(ICodeGenerator codeGenerator)
```

Comment:
>codeGenerator [](start = 62, length = 13)

Since we dereference this later without checking, please add `Requires.NotNull(codeGenerator, nameof(codeGenerator));` to this ctor.